hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataInput.java

```diff
@@ -495,6 +496,11 @@ public short readShort(int position, ByteOrder byteOrder) throws EOFException {
 
     @Override
     public String[] readUTFArray() throws IOException {
+        return readStringArray();
+    }
+
+    @Override
+    public String[] readStringArray() throws IOException {
```

Comment:
I think yes. And it seems they are missing for whole `ObjectDataInput`/`ObjectDataOutput` interfaces.

Reply:
Note that `readUTF` can not be marked with @Nullable since it is already @Nonnull on the `DataInput`. Our implementation does not conform with the `DataInput` API .

Reply:
I just put @nullable to readUTF/writeUTF. Even tough IDE complained checkstyle and spotbugs did not. So I will ignore the IDE complaint and put the annotation.